crates/rust-analyzer/src/handlers/request.rs

```diff
@@ -1055,6 +1055,7 @@ pub(crate) fn handle_references(
     let position = from_proto::file_position(&snap, params.text_document_position)?;
 
     let exclude_imports = snap.config.find_all_refs_exclude_imports();
+    let exclude_tests = snap.config.find_all_refs_exclude_tests();
```

Comment:
exclude_tests should be pushed down as a parameter to find_all_refs.

that way, a potentially more optimal implementation would be able to avoid even analyzing the test, which would be faster than filtering out results after the fact.

I think the second parameter we are passing here, which is currently None, is something like a SearchScope. It exists exactly to narrow down search results, and I think could be extended to exclude tests.

Reply:
good point, I had create an [issue](https://github.com/rust-lang/rust-analyzer/issues/17184) for this

(sorry for the slow response, kind of busy recently)